src/main/java/ui/MenuControl.java

```diff
@@ -151,9 +170,15 @@ private void onBoardSave() {
         Optional<String> response = dlg.showAndWait();
 
         if (response.isPresent()) {
-            prefs.addBoard(response.get(), panels);
+            String boardName = response.get();
+            if (boardName.equals("")) {
+                logger.info("Did not save new board");
+                return;
+            }
```

Comment:
You might want to factor the validation out into another method, and log each case, for example if it's empty, if it consists only of whitespace, etc.